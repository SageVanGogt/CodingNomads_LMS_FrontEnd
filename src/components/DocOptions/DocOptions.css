.DocOptions_select {
  width: 240px;
  height: 20px;
  margin-top: 10px;
  font-size: .9rem; }

.DocOptions_delete {
  background-color: transparent;
  height: 15px; }
