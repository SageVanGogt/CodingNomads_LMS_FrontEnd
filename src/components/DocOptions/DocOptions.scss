.DocOptions_select {
  width: 240px;
  height: 30px;
  margin-top: 10px;
  font-size: 1.2rem;
}

.DocOptions_delete {
  background-color: transparent;
  height: 25px;
}
