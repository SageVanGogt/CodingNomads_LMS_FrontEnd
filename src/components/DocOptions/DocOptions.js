import React from 'react';
import './DocOptions.css';
import PropTypes from 'prop-types';

export const DocOptions = ({ docs, handleSelectDoc, deleteDoc, id, docSelected }) => {
  const docOptions = docs.map((doc, index) => {
    return (
      <option
        key={`doc-${index}`}
        name="documentation"
        value={doc.id}
      >
        {doc.topic}
      </option>
    );
  });

  return (
    <div>
      <select
        name=""
        className="DocOptions_select"
        value={docSelected && docSelected.id}
        onChange={(event) => handleSelectDoc(event)}
      >
        <option value='' selected></option>
        { docOptions }
      </select>
      <img
        src="/button.png"
        className="DocOptions_delete"
        onClick={(event) => deleteDoc(event, id)}
      />
    </div>
  );
};

DocOptions.propTypes = {
  handleSelectDoc: PropTypes.func,
  docs: PropTypes.array
};
