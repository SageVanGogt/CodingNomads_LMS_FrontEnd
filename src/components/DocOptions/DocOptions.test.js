import React from 'react';
import { shallow } from 'enzyme';
import { DocOptions } from './DocOptions';

describe('DocOptions', () => {
  it('matches the snapshot if there is a selectedDoc', () => {
    const mockProps = {
      docs: [],
      handleSelectDoc: () => {},
      deleteDoc: () => {},
      id: 1,
      docSelected: { id: 1 }
    }
    const wrapper = shallow(<DocOptions { ...mockProps } />)

    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot if there is not a selectedDoc', () => {
    const mockProps = {
      docs: [],
      handleSelectDoc: () => {},
      deleteDoc: () => {},
      id: 1,
    }
    const wrapper = shallow(<DocOptions { ...mockProps } />)

    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot if a array of docs is passed in', () => {
    const mockProps = {
      docs: [{ id: 1, topic: 'doc topic' }],
      handleSelectDoc: () => {},
      deleteDoc: () => {},
      id: 1,
      docSelected: { id: 1 }
    }
    const wrapper = shallow(<DocOptions { ...mockProps } />)

    expect(wrapper).toMatchSnapshot();
  })

})
