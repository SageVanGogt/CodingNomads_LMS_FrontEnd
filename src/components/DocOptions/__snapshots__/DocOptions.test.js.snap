// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocOptions matches the snapshot if a array of docs is passed in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    deleteDoc={[Function]}
    docSelected={
        Object {
            "id": 1,
          }
    }
    docs={
        Array [
            Object {
              "id": 1,
              "topic": "doc topic",
            },
          ]
    }
    handleSelectDoc={[Function]}
    id={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <select
          className="DocOptions_select"
          name=""
          onChange={[Function]}
          value={1}
>
          <option
                    selected={true}
                    value=""
          />
          <option
                    name="documentation"
                    value={1}
          >
                    doc topic
          </option>
</select>,
        <button
          className="DocOptions_delete"
          onClick={[Function]}
>
          X
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              selected={true}
              value=""
/>,
            Array [
              <option
                name="documentation"
                value={1}
>
                doc topic
</option>,
            ],
          ],
          "className": "DocOptions_select",
          "name": "",
          "onChange": [Function],
          "value": 1,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "selected": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "doc-0",
            "nodeType": "host",
            "props": Object {
              "children": "doc topic",
              "name": "documentation",
              "value": 1,
            },
            "ref": null,
            "rendered": "doc topic",
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "DocOptions_delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select
            className="DocOptions_select"
            name=""
            onChange={[Function]}
            value={1}
>
            <option
                        selected={true}
                        value=""
            />
            <option
                        name="documentation"
                        value={1}
            >
                        doc topic
            </option>
</select>,
          <button
            className="DocOptions_delete"
            onClick={[Function]}
>
            X
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                selected={true}
                value=""
/>,
              Array [
                <option
                  name="documentation"
                  value={1}
>
                  doc topic
</option>,
              ],
            ],
            "className": "DocOptions_select",
            "name": "",
            "onChange": [Function],
            "value": 1,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "selected": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "doc-0",
              "nodeType": "host",
              "props": Object {
                "children": "doc topic",
                "name": "documentation",
                "value": 1,
              },
              "ref": null,
              "rendered": "doc topic",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "DocOptions_delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`DocOptions matches the snapshot if there is a selectedDoc 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    deleteDoc={[Function]}
    docSelected={
        Object {
            "id": 1,
          }
    }
    docs={Array []}
    handleSelectDoc={[Function]}
    id={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <select
          className="DocOptions_select"
          name=""
          onChange={[Function]}
          value={1}
>
          <option
                    selected={true}
                    value=""
          />
</select>,
        <button
          className="DocOptions_delete"
          onClick={[Function]}
>
          X
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              selected={true}
              value=""
/>,
            Array [],
          ],
          "className": "DocOptions_select",
          "name": "",
          "onChange": [Function],
          "value": 1,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "selected": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "DocOptions_delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select
            className="DocOptions_select"
            name=""
            onChange={[Function]}
            value={1}
>
            <option
                        selected={true}
                        value=""
            />
</select>,
          <button
            className="DocOptions_delete"
            onClick={[Function]}
>
            X
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                selected={true}
                value=""
/>,
              Array [],
            ],
            "className": "DocOptions_select",
            "name": "",
            "onChange": [Function],
            "value": 1,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "selected": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "DocOptions_delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`DocOptions matches the snapshot if there is not a selectedDoc 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    deleteDoc={[Function]}
    docs={Array []}
    handleSelectDoc={[Function]}
    id={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <select
          className="DocOptions_select"
          name=""
          onChange={[Function]}
          value={undefined}
>
          <option
                    selected={true}
                    value=""
          />
</select>,
        <button
          className="DocOptions_delete"
          onClick={[Function]}
>
          X
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              selected={true}
              value=""
/>,
            Array [],
          ],
          "className": "DocOptions_select",
          "name": "",
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "selected": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "DocOptions_delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select
            className="DocOptions_select"
            name=""
            onChange={[Function]}
            value={undefined}
>
            <option
                        selected={true}
                        value=""
            />
</select>,
          <button
            className="DocOptions_delete"
            onClick={[Function]}
>
            X
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                selected={true}
                value=""
/>,
              Array [],
            ],
            "className": "DocOptions_select",
            "name": "",
            "onChange": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "selected": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "DocOptions_delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
